### ISSUE TITLE
Option to "Show" Splash Screen

### ISSUE THREAD

**Issue by alexborton** [zoontek/react-native-bootsplash#455]
### Why it is needed?

Given a custom CodePush flow that leads to users restarting the app before they continue their journey, it would be really nice to have to option to "show" the splash screen at a point in time (and hide it again on request). This would hopefully cover the flash of "root view" that occurs between JS reloads when restarting a foreground app.

**Comment by zoontek** [comment]
Hello 👋

I recommend you to perform a quick search? https://github.com/zoontek/react-native-bootsplash/issues?q=label%3A%22feature+request%22+is%3Aclosed+show

**Comment by viLeeKorn** [comment]
@zoontek could you provide the access for me to the link as well?

**Comment by zoontek** [comment]
@viLeeKorn I fixed it. It's just a link to the existing, similar issues where you can easily find more infos.

**Comment by Melekhin** [comment]
> I fixed it. It's just a link to the existing, similar issues where you can easily find more infos.

Hello!
So at the moment is there an option to manually show the splash screen?

React Native 70+

**Comment by zoontek** [comment]
@Melekhin Hello! No. But I think this is not a good idea anyway.

A way better solution would be a package allowing you to render any other react native app root inside a controllable `UIViewController` / `DialogFragment`. This way, people would be able to render whatever they want (ex: a lock illustration when the app is in multitask because your app has sensitive data, a screen telling the user that you perform an OTA when using Codepush, etc). Just using the splash screen for that works, but it's IMHO a poor solution.

**Comment by Melekhin** [comment]
@zoontek 

Your suggestion sounds very sensible, and will satisfy all the tasks users face, but unfortunately there is no such package yet, a simple solution is to show at least just a normal boot static screen at startup....

By the way, the problem with code push and displaying a picture is also relevant to me.

If someone will start to do such a work on the package, perhaps joined the work on it myself, if I can help in any way)

**Comment by zoontek** [comment]
@Melekhin You can also create this package 🙂

**Comment by a-eid** [comment]
@zoontek if you use expo, I believe expo-splash-screen does this by default, whenever you reload the app ( in cases like OTA or RTL / LTR layout changes ) it displays the splash screen. 

@zoontek  is it technically possible for bootsplash to support this default behavior as well ?

**Comment by zoontek** [comment]
@a-eid Check the "What's next?" section here: https://github.com/zoontek/react-native-bootsplash/pull/475

**Comment by Vadko** [comment]
any progress on library for different cover screens?

**Comment by zoontek** [comment]
@Vadko No, I gave up on this idea. A few companies reach me by email but after asking, none wanted to sponsor it.

**Comment by Kasendwa** [comment]
> [@Vadko](https://github.com/Vadko) No, I gave up on this idea. A few companies reached me by email but after asking, none wanted to sponsor it.

I wonder what sponsoring such would look like cost-wise. May be a couple of us can work towards sponsoring it.

**Comment by 2winter-dev** [comment]
Does it support setting custom splash images?
